polygeist/tools/cgeist/Test/Verification/vecsplat.c

```diff
@@ -0,0 +1,39 @@
+// RUN: cgeist %s --function=* -S | FileCheck %s
+// RUN: cgeist %s --function=* -S -emit-llvm | FileCheck %s --check-prefix=CHECK-LLVM
+
+typedef int int_t_vec __attribute__((ext_vector_type(3)));
+typedef float float_t_vec __attribute__((ext_vector_type(4)));
+
+// CHECK-LABEL:   func.func @splat_int(
+// CHECK-SAME:                         %[[VAL_0:.*]]: i32) -> vector<3xi32>
+// CHECK-NEXT:      %[[VAL_1:.*]] = vector.splat %[[VAL_0]] : vector<3xi32>
+// CHECK-NEXT:      return %[[VAL_1]] : vector<3xi32>
+// CHECK-NEXT:    }
+
+// CHECK-LLVM-LABEL:    define <3 x i32> @splat_int(
+// CHECK-LLVM-SAME:                                 i32 %[[VAL_0:.*]]) {
+// CHECK-LLVM-NEXT:       %[[VAL_1:.*]] = insertelement <3 x i32> undef, i32 %[[VAL_0]], i32 0
```

Comment:
If I use clang I see that instead of `undef` it uses `poison`. Can we do the same?

Reply:
I haven't noticed that! That's a good point. This comes from the `vector.splat` lowering. I'd argue we should not modify this (in this PR). Can I simply create a PR upstream to change how this operation is lowered to address this?

Reply:
**UPDATE**: I don't think there is a way to introduce this in MLIR yet.

Reply:
Interesting.  So for now using undef is OK. We can revisit this once there is a way to represent poison in MLIR.